crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -76,12 +78,11 @@ pub fn collect_function_level_profiling_info(
     // profile tree. It is because technically they don't belong to any function, but still increase
     // the number of total steps. The value is different from zero only for functions run with header.
     let mut header_steps = Steps(0);
-    let mut current_function_steps = Steps(0);
     let mut end_of_program_reached = false;
 
     for step in trace {
-        // Skip the header. This only makes sense when a header was added to CASM program.
-        if step.pc < real_minimal_pc && run_with_call_header {
```

Comment:
Unrelated change, but this check was redundant (pc cannot be less than 1 anyways)